lib/driver.js

```diff
@@ -484,6 +482,52 @@ export class XCUITestDriver extends BaseDriver {
     }
   }
 
+  /**
+   * Handles MJPEG server-related capabilities
+   * @returns {Promise<void>}
+   */
+  async handleMjpegOptions() {
+    await this.allocateMjpegServerPort();
+    // turn on mjpeg stream reading if requested
+    if (this.opts.mjpegScreenshotUrl) {
+      this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
+      this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
+      await this.mjpegStream.start();
+    }
+  }
+
+  /**
+   * Allocates and configures port forwarding for the MJPEG server
+   * @returns {Promise<void>}
+   * @throws {Error} If port forwarding fails
+   */
+  async allocateMjpegServerPort() {
+    const mjpegServerPort = this.opts.mjpegServerPort || DEFAULT_MJPEG_SERVER_PORT;
+    this.log.debug(
+      `Forwarding MJPEG server port ${mjpegServerPort} to local port ${mjpegServerPort}`,
+    );
+    try {
+      await DEVICE_CONNECTIONS_FACTORY.requestConnection(this.opts.udid, mjpegServerPort, {
+        devicePort: mjpegServerPort,
+        usePortForwarding: this.isRealDevice(),
```

Comment:
In the screenshot recorder we also used `isLocalHost(wdaBaseUrl)` as the additional condition to determine if port forward is needed. The reason is that port forward will most likely fail if WDA is deployed on a remote device, which is most likely not even available via USB connection. Nevertheless I believe the error handling logic below should cover this case